frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.tsx

```diff
@@ -71,29 +127,79 @@ export const AccessControlTab = ({ org, app }: AccessControlTabProps): ReactNode
       }
       case 'success': {
         const currentPartyTypesAllowed = appMetadata?.partyTypesAllowed ?? initialPartyTypes;
+
         return (
-          <Checkbox.Group
-            legend={t('settings_modal.access_control_tab_checkbox_legend')}
-            size='small'
-            onChange={(newValues: string[]) => handleChange(newValues, currentPartyTypesAllowed)}
-            value={Object.keys(currentPartyTypesAllowed).filter(
-              (key) => currentPartyTypesAllowed[key],
-            )}
-          >
-            <Paragraph asChild size='small' short className={classes.checkboxParagraph}>
+          <>
+            <TabHeader
+              text={t('settings_modal.access_control_tab_checkbox_legend_label')}
+            ></TabHeader>
+            <Paragraph size='medium'>
               <span>{t('settings_modal.access_control_tab_checkbox_description')}</span>
             </Paragraph>
-            {displayCheckboxes()}
-          </Checkbox.Group>
+
+            <Table className={classes.tableContent}>
+              <Table.Head>
+                <Table.Row>
+                  <Table.HeaderCell className={classes.header}>
+                    <Checkbox
+                      aria-label={t('settings_modal.access_control_tab_option_all_type_partner')}
+                      indeterminate={tableHeaderCheckboxState === CheckboxState.Indeterminate}
+                      checked={tableHeaderCheckboxState === CheckboxState.Checked}
+                      onChange={handleTableHeaderCheckboxChange}
+                      aria-checked
+                      size='small'
+                      value='all'
+                    />
+                  </Table.HeaderCell>
+                  <Table.HeaderCell className={classes.header} aria-hidden>
+                    {t('settings_modal.access_control_tab_option_all_type_partner')}
+                  </Table.HeaderCell>
+                </Table.Row>
+              </Table.Head>
+
+              <Table.Body>
+                {getPartyTypesAllowedOptions().map((option, key) => (
+                  <Table.Row key={option.value}>
+                    <Table.Cell className={classes.checkboxContent}>
+                      <Checkbox
+                        key={key}
+                        onChange={() =>
+                          handleChange(checkedCheckboxes, currentPartyTypesAllowed, option.value)
+                        }
+                        size='small'
+                        value={option.value}
+                        checked={checkedCheckboxes.includes(option.value)}
+                      />
+                      {renderModal()}
+                    </Table.Cell>
+                    <Table.Cell>{t(option.label)}</Table.Cell>
+                  </Table.Row>
+                ))}
+              </Table.Body>
+            </Table>
+          </>
```

Comment:
Can we maybe create a separate component (and file) for this? `SelectAllowedPartyTypes` maybe?

Reply:
fixed